src/main/java/com/corundumstudio/socketio/namespace/Namespace.java

```diff
@@ -176,9 +176,9 @@ public void onDisconnect(SocketIOClient client) {
         allClients.remove(client.getSessionId());
 
         // client must leave all rooms and publish the leave msg one by one on disconnect.
+        storeFactory.pubSubStore().publish(PubSubType.BULK_LEAVE, new BulkJoinLeaveMessage(client.getSessionId(), joinedRooms, getName()));
```

Comment:
I think it should be called after all cleanup. At 184 line

Reply:
this set gets modified when `leave` is called, I can make a copy of this set, and do a bulk leave using that copy at #184, does that sound good

Reply:
apologies for not following up here, have pushed a commit with the above change, let me know if this looks good